v4.7.0 Desktop App changelog (#4402)
mattermost/docs: source/help/apps/desktop-changelog.rst

Desktop Application Changelog
========================================

Release v4.7
----------------------------

**Download Binaries:** `Mattermost Desktop on GitHub <https://github.com/mattermost/desktop/releases/latest>`_

**Release day TBD**

Breaking Changes or Upgrade Notes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
- 

Highlights
~~~~~~~~~~~~~~~

- Added support for Electron BrowserView, an underlying architecture change that improves performance and offers snappier interactions (i.e., less lag), lower CPU usage, and faster launch times.

Improvements
~~~~~~~~~~~~~~~

Windows
^^^^^^^^^^^^^
- Windows desktop now automatically switches between light and dark themes based on the operating system settings.

All Platforms
^^^^^^^^^^^^^
- Added a setting to specify the default desktop app download location.
- Improved the launch screen and loading indicator.

> - Restored deeplinking.

Reviewer: Was this a bug fix?

- Improved the spell check dictionary to provide more accurate spelling suggestions in more languages. The spell check language is now automatically based on the operating system setting.

> - Added improvements to be consistent on the use of URL and URL libraries.

Reviewer 2: ```suggestion
- Added improvements to be consistent with the use of URL and URL libraries.
```

- Ctrl/CMD + F functionality has been replaced with in-channel search (requires Mattermost server v5.35+).
- Updated the Content Security Policy for Desktop App to avoid warnings in the dev tools.
- Removed the menu bar in the Settings window for Windows and Linux.
- Shortened the max-length/width for server tab names to 224px.

> - Udpated the menubar and system tray icons for improved contrast.

Reviewer 2: ```suggestion
- Updated the menu bar and system tray icons for improved contrast.
```

> - Removed ``libappnotify1`` as a dependency requirement in Debian installers as it is no longer shipped in Debian's Bullseye. It is still recommeded to install where available.

Reviewer 2: ```suggestion
- Removed ``libappnotify1`` as a dependency requirement in Debian installers as it's no longer shipped in Debian's Bullseye. It's still recommended to install where available.
```

Architectural Changes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> - Major version upgrade of Electron to v11.4.4. Electron is the underlying technology used to build the Desktop apps.

Reviewer 2: ```suggestion
- Major version upgrade of Electron to v11.4.4. Electron is the underlying technology used to build the Desktop app.
```

- Added support for Electron BrowserView.
- Added support for M1 architecture (beta) in the build pipeline.

Bug Fixes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Windows
^^^^^^^^^^^^^
- Fixed an issue where Windows desktop notifications did not auto-dismiss when another notification arrived.
- Fixed an issue on Windows where the **Pin to Taskbar** icon got lost during an upgrade.
- Fixed an issue with the MSI build that caused notifications to not open the application and navigate to the correct channel.

MacOS
^^^^^^^^^^^^^
- Fixed an issue where changing the theme from the **System Preferences** changed the tray icon, but the red/blue dot indicating unreads got removed.
- Fixed an issue where there was an invisible Mattermost icon in the top menu bar.

All Platforms
^^^^^^^^^^^^^
- Fixed an issue where special characters were not shown for server names using GPO.
- Fixed an issue where the close/back button in permanent link media previews was missing.
- Fixed an issue where the text input focus was lost when closing the **Settings** window.
- Fixed an issue where saving the desktop app settings didn't remove the **saving** indicator in the settings window.

> - Fixed an issue where the number of mentions jewel were not shown in the tab.

Reviewer 2: ```suggestion
- Fixed an issue where the jewel indicating the number of mentions was not shown in the tab.
```

> - Fixed an issue where the desktop linting didn't match with the webapp one.

Reviewer 2: ```suggestion
- Fixed an issue where the desktop linting didn't match the webapp linting.
```

Desktop Application Changelog
========================================

Release v4.7
----------------------------

**Download Binaries:** `Mattermost Desktop on GitHub <https://github.com/mattermost/desktop/releases/latest>`_

**Release day TBD**

Breaking Changes or Upgrade Notes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
- 

Highlights
~~~~~~~~~~~~~~~

> - Added support for BrowserView.

Reviewer: Added support for Electron BrowserView, an underlying architecture change that improves performance and offers snappier interactions (i.e., less lag), lower CPU usage, and faster launch times.

Improvements
~~~~~~~~~~~~~~~

Windows
^^^^^^^^^^^^^

> - Removed the dark mode toggle option for Windows.

Reviewer 3: We should specify that dark mode now is controlled by the OS's default dark mode setting, so it doesn't look like we've removed functionality

Reviewer: @amyblais "Windows desktop now automatically switches between light and dark themes based on the operating system settings."

MacOS
^^^^^^^^^^^^^
-

All Platforms
^^^^^^^^^^^^^

> - Added Swedish language to the spellchecker.

Reviewer: Does this still apply with the Electron spellchecker?

Reviewer 4: no

Author: Removed.

- Added a setting to specify the default desktop app download location.

> - Updated the desktop app loading indicator.

Reviewer: "Improved launch screen and loading indicator"

- Restored deeplinking.

> - Switched the dictionary to the one Electron uses to have the ability to fix spelling.

Reviewer: @amyblais lets please write these in terms of the user benefit, for example:

- Improved the spell check dictionary to provide more accurate spelling suggestions in more languages. The spell check language is now automatically based on the operating system setting.

- Added improvements to be consistent on the use of URL and URL libraries.

> - Removed the Finder functionality from the Desktop App.

Reviewer: We should never have "removed" in the improvements section. Propose:

Finder functionality has been replaced with in-channel search when using the Mattermost server v5.35 and later.

> - Updated CSP for Desktop App to avoid warnings in the dev tools.

Reviewer: CSP?

Author: Maybe Guillermo can clarify.

Reviewer 4: Content Security Policy, Deving worked on improving some of the segurity mechanisms recommended for electron apps

> - Removed the menu bar in the Settings window for Windows and Linux.

Reviewer: Not sure what this is

Author: https://github.com/mattermost/desktop/pull/1530

Author: @Willyfrog Would you have details on how to describe this better? Was it a bug fix?

Author: @Willyfrog ^

Reviewer 4: we can remove that from the list of changes, the actual change is that we now have the settings in a separate window. the PR mentions that we removed that because in linux and windows each window has its own menu window, while in mac it is a shared one

- Shortened the max-length/width for server tab names to 224px.

> - For Windows and Linux, the system language is now used and the spell checker's option help text was improved. languages are no longer selectable and toggling the spellchecker requires a restart.

Reviewer: Remove as it's covered by my suggestion above

- Udpated the menubar and system tray icons for improved contrast.
- Removed ``libappnotify1`` as a dependency requirement in Debian installers as it is no longer shipped in Debian's Bullseye. It is still recommeded to install where available.

Architectural Changes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> 

Reviewer 3: I wonder if it makes sense to highlight BrowserView under this section

- Major version upgrade of Electron to v11.4.4. Electron is the underlying technology used to build the Desktop apps.
- Added support for M1 architecture in the build pipeline.

Bug Fixes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Windows
^^^^^^^^^^^^^
- Fixed an issue where Windows desktop notifications did not auto-dismiss when another notification arrived.
- Fixed an issue on Windows where the **Pin to Taskbar** icon got lost during an upgrade.

> 

Reviewer 3: I'll have one note to add from this PR: https://github.com/mattermost/desktop/pull/1590

MacOS
^^^^^^^^^^^^^
- Fixed an issue where changing the theme from the **System Preferences** changed the tray icon, but the red/blue dot indicating unreads got removed.
- Fixed an issue where there was an invisible Mattermost icon in the top menu bar.

All Platforms
^^^^^^^^^^^^^
- Fixed an issue where special characters were not shown for server names using GPO.
- Fixed an issue where the close/back button in permanent link media previews was missing.
- Fixed an issue where the text input focus was lost when closing the **Settings** window.
- Fixed an issue where saving the desktop app settings didn't remove the **saving** indicator in the settings window.
- Fixed an issue where the number of mentions jewel were not shown in the tab.
- Fixed an issue where the desktop linting didn't match with the webapp one.
- Fixed an issue where clicking on a notification did nothing when the wrong server tab was selected.
- Fixed an issue where users were unable to copy text from desktop **About** window.

Known Issues
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> - An error may occur when installing the MSI Installer on any Windows version.

Reviewer 3: @Willyfrog is this still a thing?

- Added a setting to specify the default desktop app download location.
- Improved the launch screen and loading indicator.
- Restored deeplinking.
- Improved the spell check dictionary to provide more accurate spelling suggestions in more languages. The spell check language is now automatically based on the operating system setting.
- Added improvements to be consistent on the use of URL and URL libraries.
- Ctrl/CMD + F functionality has been replaced with in-channel search (requires Mattermost server v5.35+).
- Updated CSP for Desktop App to avoid warnings in the dev tools.
- Removed the menu bar in the Settings window for Windows and Linux.
- Shortened the max-length/width for server tab names to 224px.
- Udpated the menubar and system tray icons for improved contrast.
- Removed ``libappnotify1`` as a dependency requirement in Debian installers as it is no longer shipped in Debian's Bullseye. It is still recommeded to install where available.

Architectural Changes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

- Major version upgrade of Electron to v11.4.4. Electron is the underlying technology used to build the Desktop apps.
- Added support for Electron BrowserView.

> - Added support for M1 architecture in the build pipeline.

Reviewer 4: I'd highlight M1 as beta for a while

Architectural Changes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> - Major version upgrade of Electron to v12.0.0. Electron is the underlying technology used to build the Desktop app.

Reviewer 3: More accurately this should be v12.0.7 (if that matters)
